src/main/java/ru/sbt/mipt/oop/DoorEventManager.java

```diff
@@ -0,0 +1,27 @@
+package ru.sbt.mipt.oop;
+
+import static ru.sbt.mipt.oop.LightEventManager.turnOffAllLights;
+import static ru.sbt.mipt.oop.SensorEventType.DOOR_OPEN;
+
+public class DoorEventManager {
+    static void manageDoorEvent(SmartHome smartHome, SensorEvent event) {
+        for (Room room : smartHome.getRooms()) {
+            for (Door door : room.getDoors()) {
+                if (door.getId().equals(event.getObjectId())) {
+                    if (event.getType() == DOOR_OPEN) {
+                        door.setOpen(true);
+                        System.out.println("Door " + door.getId() + " in room " + room.getName() + " was opened.");
+                    } else {
+                        door.setOpen(false);
+                        System.out.println("Door " + door.getId() + " in room " + room.getName() + " was closed.");
+                        // если мы получили событие о закрытие двери в холле - это значит, что была закрыта входная дверь.
+                        // в этом случае мы хотим автоматически выключить свет во всем доме (это же умный дом!)
+                        if (room.getName().equals("hall")) {
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D1%80%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D1%8F%D0%B9%D1%82%D0%B5-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BE%D0%B2-%D0%B8-%D1%81%D1%86%D0%B5%D0%BD%D0%B0%D1%80%D0%B8%D0%B8-%D1%83%D0%BC%D0%BD%D0%BE%D0%B3%D0%BE-%D0%B4%D0%BE%D0%BC%D0%B0